docs/reference/builder.md

```diff
@@ -180,21 +180,28 @@ that set `abc` to `bye`.
 
 ### .dockerignore file
 
-If a file named `.dockerignore` exists in the root of `PATH`, then Docker
-interprets it as a newline-separated list of exclusion patterns. Docker excludes
-files or directories relative to `PATH` that match these exclusion patterns. If
-there are any `.dockerignore` files in `PATH` subdirectories, Docker treats
-them as normal files. 
-
-Filepaths in `.dockerignore` are absolute with the current directory as the
-root. Wildcards are allowed but the search is not recursive. Globbing (file name
-expansion) is done using Go's
-[filepath.Match](http://golang.org/pkg/path/filepath#Match) rules.
-
-You can specify exceptions to exclusion rules. To do this, simply prefix a
-pattern with an `!` (exclamation mark) in the same way you would in a
-`.gitignore` file. Currently there is no support for regular expressions.
-Formats like `[^temp*]` are ignored. 
+If a file named `.dockerignore` exists at the root of the source
+repository, Docker interprets it as a newline-separated list of exclusion
+patterns. Docker excludes from the build context files or directories whose
+paths relative to the root of the source repository match these exclusion
+patterns. If there are any files named `.dockerignore` elsewhere in the
+source repository, Docker treats them as normal files.
+
+Matching is done using Go's
+[filepath.Match](http://golang.org/pkg/path/filepath#Match) rules,
+which are similar to those of Unix shells (but also different).  A
```

Comment:
OK, fair enough, I will take that out.  I think I was a little shocked by the `filepath#Match` rules is all.  If I am understanding them correctly, they don't treat the path components `.` and `..` specially.  (The first will be matched by `*` and `?`, and the second will be matched by `*` and `??`.)  Also, I believe that with shell globbing, no pattern that doesn't contain a `/` can match a `/`, but with Go you can match a `/` with a character range like `[.-0]`.